java/src/test/java/ai/rapids/cudf/BinaryOpTest.java

```diff
@@ -136,6 +136,40 @@ public static <S> ColumnVector forEachS(DType retType, S lhs, ColumnVector rhs,
     }
   }
 
+  public static ColumnVector forEachEvalNulls(DType retType, ColumnVector lhs, ColumnVector rhs, CpuOpVV op) {
+    int len = (int)lhs.getRowCount();
+    try (HostColumnVector hostLHS  = lhs.copyToHost();
+         HostColumnVector hostRHS = rhs.copyToHost();
+         Builder builder = HostColumnVector.builder(retType, len)) {
+      for (int i = 0; i < len; i++) {
+        op.computeNullSafe(builder, hostLHS, hostRHS, i);
+      }
+      return builder.buildAndPutOnDevice();
+    }
+  }
+
+  public static <S> ColumnVector forEachSEvalNulls(DType retType, ColumnVector lhs, S rhs, CpuOpVS<S> op) {
```

Comment:
nit: can this be named better?

Reply:
I reworked it slightly and now take `evalNulls` as an optional argument. I think it's around the cleaner solution.